model/user.go

```diff
@@ -173,3 +174,56 @@ func GetUserScore(username string) (score UserScore, err error) {
 	`, username).Scan(&score.Score)
 	return
 }
+
+// Friend holds a friend
+type Friend struct {
+	Username   string `json:"username"`
+	CreateTime string `json:"createTime"`
+	Friend     string `json:"friend"`
+}
+
+// GetFriends returns
+func GetFriends(identifier string) (friends []Friend, err error) {
+	friendRows, err := database.Query(`
+    SELECT username, create_time, friend
+    FROM "roaster"."user_friends"
+    WHERE (LOWER(username)=LOWER(TRIM($1)))
+    `, identifier)
+	if err != nil {
+		return
+	}
+	defer friendRows.Close()
+
+	for friendRows.Next() {
+		res := Friend{}
+		err = friendRows.Scan(&res.Username, &res.CreateTime, &res.Friend)
+		friends = append(friends, res)
+		if err != nil {
+			return
+		}
+	}
+	return
+}
+
+// PutFriend returns
+func PutFriend(identifier string, friend string) (err error) {
+	_, err = database.Exec(`
+    INSERT INTO "roaster"."user_friends"
+    (username, create_time, friend)
+    VALUES
+    (TRIM($1), $2, TRIM($3))
+    `, identifier, time.Now(), friend)
+	return
+}
+
+// RemoveFriend returns
+func RemoveFriend(identifier string, friend string) (err error) {
```

Comment:
`identifier string` is always the username, so maybe `username string` is a more fitting name?